source/extensions/filters/common/fault/fault_config.cc

```diff
@@ -27,17 +27,22 @@ FaultAbortConfig::FaultAbortConfig(
 }
 
 absl::optional<Http::Code>
-FaultAbortConfig::HeaderAbortProvider::status_code(const Http::HeaderEntry* header) const {
+FaultAbortConfig::HeaderAbortProvider::statusCode(const Http::HeaderEntry* header) const {
+  absl::optional<Http::Code> ret;
   if (header == nullptr) {
-    return absl::nullopt;
+    return ret;
   }
 
-  uint64_t value;
-  if (!absl::SimpleAtoi(header->value().getStringView(), &value)) {
-    return absl::nullopt;
+  uint64_t code;
+  if (!absl::SimpleAtoi(header->value().getStringView(), &code)) {
+    return ret;
+  }
+
+  if (code >= 200 and code < 600) {
```

Comment:
nit: s/and/&& (some C++ compilers don't support it and it's not widely used), also please add some out of range tests.